src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -231,6 +231,27 @@ def __lt__(self, other: Proposal) -> bool:
             self.priority == other.priority and self.source_id < other.source_id
         )
 
+    def __eq__(self, other: object) -> bool:
+        """Check if two proposals are equal.
```

Comment:
Even though I think I get the intention of this, to me the definition of equality does not feel intuitive.

Reply:
The `__lt__`, `__eq__` and `__hash__` methods are all defined based on just priority and source ID, because they are there to allow sorting by priority+source_id, and to remove elements/replace them based on the same key.

`Proposal` is an internal type, because users would use the `propose_charge`, etc. methods.

For the `SortedSet`, we needed this definition of `__lt__`, and ff we were using a `deque`, we'd need this definition of `__eq__`.  Now that we are using `set`, we need this definition of `__hash__`, but maybe the `__lt__` can be replaced by a lambda passed to the `sorted` function.  and the `__eq__` can go away.

I will try this now.

Reply:
I tried a quick test like this and removed `__lt__` and `__eq__`, but it doesn't seem to be the same.  The results are different and the tests don't pass.

```python
        for next_proposal in sorted(
            proposals,
            reverse=True,
            key=lambda proposal: (proposal.priority, proposal.source_id),
        ):
```